pkg/virt-api/api.go

```diff
@@ -785,8 +793,12 @@ func (app *virtAPIApp) startTLS(informerFactory controller.KubeInformerFactory)
 
 	// start graceful shutdown handler
 	go func() {
-		s := <-c
-		log.Log.Infof("Received signal %s, initiating graceful shutdown", s.String())
+		select {
+		case s := <-c:
+			log.Log.Infof("Received signal %s, initiating graceful shutdown", s.String())
+		case msg := <-app.reInitChan:
```

Comment:
Correct me if I'm wrong, but reinit is basically the same as shutdown.  Which means there is the possibility of downtime?  Even if there are multiple apiservers running active/active, they will probably all restart at the same time?  Obviously, we are not going to hit this condition frequently but perhaps there are less impactful ways to deal with it?

The snapshot controller has a similar problem.  It can't watch VolumeSnapshots/VolumeSnapshotClasses if the CRDs don't exist.  So it watches CRDs and creates/deletes informers dynamically.  https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/snapshot/snapshot_base.go#L84

Reply:
since kubevirt and cdi are typically installed together, and this is a one time disruption event that will occur once at install time only, the re-init logic seems like a reasonable trade off that allows us to avoid introducing more complex dynamic informer loading logic. 

like we talked about out of band, I think it would be worth investing in the dynamic informers once we have a dependency on a CRD which is not a part of the kubevirt ecosystem.